### DISCUSSION TITLE
Run git gc in the repository periodically

### DISCUSSION THREAD

**Discussion by corsac-s** [Kozea/Radicale#1384] [Ideas]
Hi,
I'm using radicale with the filesystem backend stored in a git repository. It works fine, but the repository slowly fills the filesystem with objects. I usually can save space by running `git gc` every once in a while, but it would be way easier if radicale would do it by itself.
It's actually painful to run `git gc` wheen there's no space left on the device.

**Comment by mookie-** [comment]
@corsac-s I think it's hard to do that with radicale as the git command are just executed as a hook.
My proposal would be to create a cronjob which runs this regulary like this:
```
4 2 1 * * radicale git --git-dir /path/to/radicale/git/.git gc
```

**Comment by corsac-s** [comment]
Unfortunately in my case that doesn't really work, my radicale instance runs in a container without cron, but I get your point.

**Comment by corsac-s** [comment]
But systemd runs so systemd timers work, actually

**Comment by mookie-** [comment]
That's a bit of a workaround but you could also add the gc to the hook but only run it if the last run was more than 1 month ago:
```
# UNTESTED pseudocode
old_gc=$(cat /tmp/gitgctimestamp); now=$(date +%s); diff=$((${now}-${old_gc})); if [ $diff -gt 43200 ]; then git --git dir /path/to/radicale/git/.git gc; echo $now > /tmp/gitgctimestamp; fi
